pkg/store/bucket_test.go

```diff
@@ -1054,7 +1054,7 @@ func uploadTestBlock(t testing.TB, tmpDir string, bkt objstore.Bucket, series in
 		Source:     metadata.TestSource,
 	}, nil)
 	testutil.Ok(t, err)
-	testutil.Ok(t, block.Upload(context.Background(), logger, bkt, filepath.Join(tmpDir, "tmp", id.String())))
+	testutil.Ok(t, block.Upload(context.Background(), logger, bkt, filepath.Join(tmpDir, "tmp", id.String()), metadata.NoneFunc))
```

Comment:
There might be some time for Uploader struct , but not in this PR 🤗

Reply:
Yep :+1: it is starting to get a lot of arguments so we should refactor it into something more understandable